modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -3107,7 +3110,7 @@ private void recalculateAndImproveEstimatedSize(int improve) {
             this.absoluteOffset = newOffset + oldOffset;
             adjustPosition();
         }
-
+        recalculating = false;
```

Comment:
Good question. There should be no non-recoverable exceptions in the VirtualFlow specific code. However, since IndexedCell.updateItem() and others might get called from this method, exceptions there might propagate. 
My initial thought was to fail-fast, but it is probably better to use a try/finally so that at least this method is still "working" somehow. I'll add that.